vms/platformvm/state/state.go

```diff
@@ -1785,20 +1786,80 @@ func (s *state) writeBlsKeyDiffs(height uint64, blsKeyDiffs map[ids.NodeID]*bls.
 	return nil
 }
 
+func (s *state) updateValidatorSet(
+	updateValidators bool,
+	valSetDiff map[subnetNodeKey]validatorStatusPair,
+	weightDiffs map[subnetNodeKey]*ValidatorWeightDiff,
+) error {
+	if !updateValidators {
+		return nil
+	}
+
+	for key, weightDiff := range weightDiffs {
+		var (
+			subnetID = key.subnetID
+			nodeID   = key.nodeID
+			val      = valSetDiff[key]
+			err      error
+		)
+
+		if weightDiff.Amount == 0 {
+			// No weight change to record; go to next validator.
+			continue
+		}
+
+		if weightDiff.Decrease {
+			err = s.cfg.Validators.RemoveWeight(subnetID, nodeID, weightDiff.Amount)
+		} else {
+			if val.status == added {
```

Comment:
Yes it holds. I can document it, but I think the best would be to find a data arrangement that does guarantee it. The [constructive instead of declarative idea](https://www.hillelwayne.com/post/constructive/).
Not sure how to do that while minimizing data duplication.